hooks/pre-commit

```diff
@@ -0,0 +1,8 @@
+set -ex
+
+diff=$(git diff --cached --name-only | grep '\.py$' || true)
+
+if [ ! -z "$diff" ]; then
+  black --check $diff
+  pylint $diff
```

Comment:
Note running pylint against the entire code base takes a really long time (30 ~ 1 min.)

Reply:
This hook can be enabled via:

```
git config core.hooksPath hooks
```

Reply:
It would be nice to also update https://github.com/mlflow/mlflow/blob/master/CONTRIBUTING.rst
> For convenience, we provide a pre-commit git hook that validates that commits are signed-off.
to
> For convenience, we provide a pre-commit git hook that validates that commits are signed-off and auto-format the code.

Reply:
Sounds good!

Reply:
Updated!